gtk4/src/subclass/buildable.rs

```diff
@@ -0,0 +1,394 @@
+    fn parent_construct_child(
+        &self,
+        buildable: &Self::Type,
+        builder: &Builder,
+        name: &str,
+    ) -> Object;
+}
+
+impl<T: BuildableImpl> BuildableImplExt for T {
+    fn parent_set_id(&self, buildable: &Self::Type, id: &str) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Buildable>() as *const ffi::GtkBuildableIface;
+
+            let func = (*parent_iface)
+                .set_id
+                .expect("no parent \"set_id\" implementation");
+
+            func(
+                buildable.unsafe_cast_ref::<Buildable>().to_glib_none().0,
+                id.to_glib_none().0,
+            )
+        }
+    }
+
+    fn parent_id(&self, buildable: &Self::Type) -> GString {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Buildable>() as *const ffi::GtkBuildableIface;
+
+            let func = (*parent_iface)
+                .get_id
+                .expect("no parent \"get_id\" implementation");
+
+            from_glib_none(func(
+                buildable.unsafe_cast_ref::<Buildable>().to_glib_none().0,
+            ))
+        }
+    }
+
+    fn parent_add_child(
+        &self,
+        buildable: &Self::Type,
+        builder: &Builder,
+        child: &Object,
+        type_: Option<&str>,
+    ) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Buildable>() as *const ffi::GtkBuildableIface;
+
+            let func = (*parent_iface)
+                .add_child
+                .expect("no parent \"add_child\" implementation");
+
+            func(
+                buildable.unsafe_cast_ref::<Buildable>().to_glib_none().0,
+                builder.to_glib_none().0,
+                child.to_glib_none().0,
+                type_.to_glib_none().0,
+            )
+        }
+    }
+
+    fn parent_set_buildable_property(
+        &self,
+        buildable: &Self::Type,
+        builder: &Builder,
+        name: &str,
+        value: &Value,
+    ) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Buildable>() as *const ffi::GtkBuildableIface;
+
+            // gtk::Builder falls back to using ObjectExt::set_property if the method is not implemented
+            if let Some(func) = (*parent_iface).set_buildable_property {
+                func(
+                    buildable.unsafe_cast_ref::<Buildable>().to_glib_none().0,
+                    builder.to_glib_none().0,
+                    name.to_glib_none().0,
+                    value.to_glib_none().0,
+                )
+            } else {
+                buildable.set_property(name, value).unwrap_or_else(|err| {
```

Comment:
we need to use `set_property_from_value()` here.

Reply:
What's the difference ?:)

Reply:
`set_property()` takes a `ToValue`. It runs `value.into()`, which makes the actually set `value` into `Value<Value<T>`. Look at the implementation https://gtk-rs.org/gtk-rs-core/git/docs/glib/object/trait.ObjectExt.html#tymethod.set_property

Reply:
Oups indeed

Reply:
I noticed because of the nice error message you made :)

Reply:
See https://github.com/gtk-rs/gtk-rs-core/pull/85

Reply:
@jsparber should be fixed now